app/Models/User.php

```diff
@@ -834,11 +834,12 @@ public function isCurrentTenant(): bool
 
     /**
      * @return bool if the user needs to update their tenant status
-     * A user needs to update their tenant status if they are a tenant and their tenant_until date is in the past.
+     * A user needs to update their tenant status if they are a tenant and their tenant_until date is in the past;
+     * or if they have no status whatsoever.
      */
     public function needsUpdateTenantUntil(): bool
     {
-        return $this->isTenant() && !$this->isCurrentTenant();
+        return !$this->isCollegist() && !$this->isCurrentTenant();
```

Comment:
Won't this tigger for teachers and so on?

Reply:
That is filtered at RedirectToStatusUpdate.php:42; i.e. anyone who has any role is discarded. But you are right we should move that logic here.